test/kubemark/resources/start-kubemark-master.sh

```diff
@@ -176,39 +193,43 @@ function compute-kubelet-params {
 	echo "${params}"
 }
 
-# Creates the supervisord config file for kubelet from the exec_command ($1).
+# Creates the systemd config file for kubelet.service.
 function create-kubelet-conf() {
-	local -r name="kubelet"
-	local exec_command="$1 "
-	exec_command+=$(compute-kubelet-params)
-
-	cat >>"/etc/supervisor/conf.d/${name}.conf" <<EOF
-[program:${name}]
-command=${exec_command}
-stderr_logfile=/var/log/${name}.log
-stdout_logfile=/var/log/${name}.log
-autorestart=true
-startretries=1000000
+	local -r kubelet_bin="$1"
+	local -r kubelet_env_file="/etc/default/kubelet"
+	local -r flags=$(compute-kubelet-params)
+	echo "KUBELET_OPTS=\"${flags}\"" > "${kubelet_env_file}"
+
+	# Write the systemd service file for kubelet.
+	cat <<EOF >/etc/systemd/system/kubelet.service
+[Unit]
+Description=Kubermark kubelet
+Requires=network-online.target
+After=network-online.target
+
+[Service]
+Restart=always
+RestartSec=10
+EnvironmentFile=${kubelet_env_file}
+ExecStart=${kubelet_bin} \$KUBELET_OPTS
+
+[Install]
+WantedBy=multi-user.target
 EOF
 }
 
-# This function assembles the kubelet supervisord config file and starts it using
-# supervisorctl, on the kubemark master.
+# This function assembles the kubelet systemd service file and starts it using
+# systemctl, on the kubemark master.
 function start-kubelet {
-	# Kill any pre-existing kubelet process(es).
-	pkill kubelet
-	# Replace the builtin kubelet (if any) with the correct binary.
-	local -r builtin_kubelet="$(which kubelet)"
-	if [[ -n "${builtin_kubelet}" ]]; then
-		cp "${KUBE_BINDIR}/kubelet" "$(dirname "$builtin_kubelet")"
-	fi
+	# Create systemd config.
+	local -r kubelet_bin="/usr/bin/kubelet"
+	create-kubelet-conf "${kubelet_bin}"
 
-	# Create supervisord config for kubelet.
-	create-kubelet-conf "${KUBE_BINDIR}/kubelet"
+	# Flush iptables nat table.
```

Comment:
This is again something that I carried on from the gci configure-helper.sh and have removed it now. The nat IP table doesn't seem to do anything much significant  (and e2e-test works both with/without any rules in this table) except for allowing IP masquerading on post-routing. If you are curious, here is how the nat table looks like before flushing:

Chain PREROUTING (policy ACCEPT)
target     prot opt source               destination         
DOCKER     all  --  anywhere             anywhere             ADDRTYPE match dst-type LOCAL

Chain INPUT (policy ACCEPT)
target     prot opt source               destination         

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination         
DOCKER     all  --  anywhere            !127.0.0.0/8          ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT)
target     prot opt source               destination         
KUBE-POSTROUTING  all  --  anywhere             anywhere             /* kubernetes postrouting rules */
MASQUERADE  all  --  172.17.0.0/16        anywhere            

Chain DOCKER (2 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             anywhere            

Chain KUBE-MARK-DROP (0 references)
target     prot opt source               destination         
MARK       all  --  anywhere             anywhere             MARK or 0x8000

Chain KUBE-MARK-MASQ (0 references)
target     prot opt source               destination         
MARK       all  --  anywhere             anywhere             MARK or 0x4000

Chain KUBE-POSTROUTING (1 references)
target     prot opt source               destination         
MASQUERADE  all  --  anywhere             anywhere             /* kubernetes service traffic requiring SNAT */ mark match 0x4000/0x4000